petastorm/reader.py

```diff
@@ -378,8 +378,10 @@ def __init__(self, pyarrow_filesystem, dataset_path, schema_fields=None,
         :param filters: (List[Tuple] or List[List[Tuple]]): Standard PyArrow filters.
             These will be applied when loading the parquet file with PyArrow. More information
             here: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html
+        :param shard_seed: Random seed to shuffle row groups for data sharding. Defaults to None
         """
         self.num_epochs = num_epochs
+        self.shard_seed = shard_seed
```

Comment:
Since we use this value only from the constructor , suggest we don't turn it into a state variable, but pass it down the call-stack to the consumer. By limiting the scope of the variable, it makes the code more readable, IMO.

Reply:
sounds reasonable to me.